src/azure-cli/azure/cli/command_modules/acr/connected_registry.py

```diff
@@ -0,0 +1,494 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from enum import Enum
+from msrest.exceptions import ValidationError
+from knack.log import get_logger
+from knack.util import CLIError
+from azure.cli.core.commands import LongRunningOperation
+from azure.cli.core.commands.client_factory import get_subscription_id
+from ._client_factory import cf_acr_registries, cf_acr_tokens, cf_acr_scope_maps
+from ._utils import (
+    get_registry_by_name,
+    validate_managed_registry,
+    user_confirmation,
+    create_default_scope_map,
+    get_token_from_id,
+    get_scope_map_from_id,
+    parse_actions_from_repositories,
+    parse_repositories_from_actions,
+    parse_actions_from_gateways,
+    build_token_id
+)
+
+class ConnectedRegistryModes(Enum):
+    MIRROR = 'mirror'
+    REGISTRY = 'registry'
+
+DEFAULT_GATEWAY_SCOPE = ['config/read', 'config/write', 'message/read', 'message/write']
+REPO_SCOPES_BY_MODE = {
+    ConnectedRegistryModes.MIRROR.value: ['content/read', 'metadata/read'],
+    ConnectedRegistryModes.REGISTRY.value: ['content/read', 'content/write', 'content/delete',
+                                            'metadata/read', 'metadata/write']
+}
+SYNC_SCOPE_MAP_NAME = "{}-sync-scope-map"
+SYNC_TOKEN_NAME = "{}-sync-token"
+REPOSITORY = "repositories/"
+GATEWAY = "gateway/"
+
+logger = get_logger(__name__)
+
+
+def acr_connected_registry_create(cmd, # pylint: disable=too-many-locals, too-many-statements
+                                  client,
+                                  registry_name,
+                                  connected_registry_name,
+                                  repositories=None,
+                                  sync_token_name=None,
+                                  client_token_list=None,
+                                  resource_group_name=None,
+                                  mode=None,
+                                  parent_name=None,
+                                  sync_schedule=None,
+                                  sync_message_ttl=None,
+                                  sync_window=None,
+                                  log_level=None,
+                                  sync_audit_logs_enabled=True):
```

Comment:
should it be defaulted to False? not sure how the default values are passed to this create?

Reply:
The default value in the CLI spec is that the audit_log_status Should be Enabled.